src/util.c

```diff
@@ -905,36 +906,70 @@ int version2num(const char *version) {
     return v;
 }
 
+/* Global state. */
+static int seed_initialized = 0;
+static unsigned char seed[64]; /* 512 bit internal block size. */
+
+static void initializeRandomSeed(void) {
+    assert(seed_initialized == 0);
+    /* Initialize a seed and use SHA1 in counter mode, where we hash
+     * the same seed with a progressive counter. For the goals of this
+     * function we just need non-colliding strings, there are no
+     * cryptographic security needs. */
+    FILE *fp = fopen("/dev/urandom", "r");
+    if (fp == NULL || fread(seed, sizeof(seed), 1, fp) != 1) {
+        /* Revert to a weaker seed, and in this case reseed again
+         * at every call.*/
+        for (unsigned int j = 0; j < sizeof(seed); j++) {
+            struct timeval tv;
+            gettimeofday(&tv, NULL);
+            pid_t pid = getpid();
+            seed[j] = tv.tv_sec ^ tv.tv_usec ^ pid ^ (long)fp;
+        }
+    } else {
+        seed_initialized = 1;
+    }
+    if (fp) fclose(fp);
+}
+
+/* This function receives a string with 64 bytes encoded as 128 hexadecimal
+ * digits, and sets the 64 byte random seed. */
+void setRandomSeedCString(char *seed_str, size_t len) {
+    assert(len == sizeof(seed) * 2);
+    for (size_t i = 0; i < sizeof(seed); i++) {
+        sscanf(seed_str + (i * 2), "%02hhX", &seed[i]);
+    }
+    seed_initialized = 1;
+}
+
+/* This function returns a string with 64 bytes encoded as 128 hexadecimal
+ * digits, that represents the 64 bytes random seed. */
+char *getRandomSeedCString(size_t *len) {
+    if (!seed_initialized) {
+        initializeRandomSeed();
+    }
+
+    char *buff = zmalloc(sizeof(seed) * 2 + 1);
```

Comment:
Why not move the allocation out of this function? Then the caller can allocate it using zmalloc, malloc or stack allocated.

If this is fixed to 64 bytes seed, then add the `assert(len == sizeof(seed) * 2);` here too?

```suggestion
/* This function returns a string with 64 bytes encoded as 128 hexadecimal
 * digits, that represents the 64 bytes random seed. */
char *getRandomSeedCString(char *buff, size_t *len) {
    if (!seed_initialized) {
        initializeRandomSeed();
    }
    assert(len == sizeof(seed) * 2);
```